src/common/ceph_context.h

```diff
@@ -260,8 +261,24 @@ class CephContext {
 
   ceph::HeartbeatMap *_heartbeat_map;
 
-  ceph::spinlock _associated_objs_lock;
-  std::map<std::string, SingletonWrapper*> _associated_objs;
+  ceph::spinlock associated_objs_lock;
+
+  struct associated_objs_cmp {
+    using is_transparent = std::true_type;
+    template<typename T, typename U>
+    bool operator ()(const std::pair<T, const std::type_info*>& l,
+		     const std::pair<U, const std::type_info*>& r) const noexcept {
+      return ((l.first < r.first)  ||
+	      (l.first == r.first && l.second->before(*r.second)));
+    }
+  };
+
+  // I would use std::type_index, but the version of the library on
+  // gitbuilder doesn't support it for whatever reason.
```

Comment:
weird ... we should have `std::type_index`. could you point me to the build failure? also, we are not using gitbuilder anymore. we are using jenkins for building/testing the PRs.

Reply:
It was the first push I did for this branch. I'm not sure if they're kept. I could stick type-index back in and repush to show you.